src/modules/thirdParty/sagas.ts

```diff
@@ -41,13 +60,62 @@ export function* getContractInstance(
   return contractInstance
 }
 
+export function* getPublishItemsSignature(thirdPartyId: string, qty: number) {
+  const maticChainId: ChainId = yield call(getChainIdByNetwork, Network.MATIC)
+  const provider: Provider | null = yield call(getConnectedProvider)
+  if (!provider) {
+    throw new Error('Could not get a valid connected Wallet')
+  }
+  const thirdPartyContract: ContractData = yield call(getContract, ContractName.ThirdPartyRegistry, maticChainId)
+  const salt = utils.hexlify(utils.randomBytes(32))
+  const domain = {
+    name: thirdPartyContract.name,
+    verifyingContract: thirdPartyContract.address,
+    version: thirdPartyContract.version,
+    salt
+  }
+  const dataToSign = {
+    thirdPartyId,
+    qty,
+    salt
+  }
+  const domainTypes = {
+    ConsumeSlots: [
+      { name: 'thirdPartyId', type: 'string' },
+      { name: 'qty', type: 'uint256' },
+      { name: 'salt', type: 'bytes32' }
+    ]
+  }
+
+  // TODO: expose this as a function in decentraland-transactions
+  const msgString = JSON.stringify({ domain, message: dataToSign, types: domainTypes, primaryType: 'ConsumeSlots' })
+
+  const accounts: string[] = yield call([provider, 'request'], { method: 'eth_requestAccounts', params: [], jsonrpc: '2.0' })
+  const from = accounts[0]
+
+  const signature: string = yield call([provider, 'request'], {
+    method: 'eth_signTypedData_v4',
+    params: [from, msgString],
+    jsonrpc: '2.0'
+  })
+
```

Comment:
I want to clarify something here. There's a competing implementation between web3x and ethers in this codebase. The reason why you don't see a signer or a request here (or in the marketplace) is because you should do something like:

```
const eth: Eth = yield call(getEth)

const personal = new Personal(eth.provider)
const signature: string = yield personal.sign(...)
```

where `getEth` is a wrapper of the provider for web3x (the old lib we used to use).

I think this implementation is correct, but I wanted to clarify why this came up. An example of a signature can be seen on the ens module's handler `handleSetAlias` here in the builder